### PR TITLE
feat: add support for closestOnCircle

### PR DESCRIPTION
**Pull request by danyhoron**

### CODE REVIEW

File: src/leaflet.geometryutil.js [hunk 1]
```diff
@@ -157,6 +157,36 @@ L.GeometryUtil = L.extend(L.GeometryUtil || {}, {
...
+        const cx = center.lng;
+        const cy = center.lat;
+        const dx = x - cx;
+        const dy = y - cy;
+        const distanceSquared = dx * dx + dy * dy;
+
+        const distance = Math.sqrt(distanceSquared)
+
+        const tx = cx + (dx / distance) * radius;
+        const ty = cy + (dy / distance) * radius;
```
Comment (leplatrem) [inline src/leaflet.geometryutil.js:184]: Can you add a comment explaining what you are doing here?
(eg. cross multiplication between the distance of the point to the center and the radius)

File: spec/test.closestCircle.js [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    const circle = new L.Circle(center, { radius });
+    const closest = L.GeometryUtil.closestOnCircle(circle, point);
+    assert.equal(expected.toString(), closest.toString());
+    done();
+  });
+
+  it("It should be closest if is outside the circle", function (done) {
+    const center = L.latLng([0, 0]);
+    const radius = 5;
+    const point = L.latLng([0, 6]);
```
Comment (leplatrem) [inline spec/test.closestCircle.js:26]: nitpick: you could use a negative value here to test that it works when the point is on the left

**Review by leplatrem — APPROVED** [review]
Thank you for taking the time to contribute this! 
It looks sound, but would leave @babastienne/@bastyen formally approve and release ;)

File: src/leaflet.geometryutil.js (hunk 1 above)
Comment (danyhoron) [reply src/leaflet.geometryutil.js:186]: I added some comments

File: spec/test.closestCircle.js (hunk 2 above)
Comment (danyhoron) [reply spec/test.closestCircle.js:26]: I added a few more scenarios

**Pull request merged**